### PR TITLE
Logging state during building of TracerProvider

### LINKED ISSUE #3644: [Tracing] Improve internal logging of TracerProvider configuration
The [MeterProvider does a really good job of logging its state](https://github.com/open-telemetry/opentelemetry-dotnet/blob/872a52f5291804c7af19e90307b5cc097b2da709/src/OpenTelemetry/Metrics/MeterProviderSdk.cs#L96) when it is constructed. We should do the same in the TracerProvider[Sdk]. The idea is this will help troubleshoot configuration issues.

### PR DESCRIPTION
**Pull request by ggoel**
Fixes #3644

## Changes

Logging state during building of TracerProvider.

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by utpilla** [comment]
@ggoel Thanks for adding diagnostic logging for TracerProviderSdk. The PR looks good. Could you please fix the build errors?

**Review by utpilla — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

File: src/OpenTelemetry/Trace/TracerProviderSdk.cs [hunk 1]
```diff
@@ -49,12 +50,17 @@ internal sealed class TracerProviderSdk : TracerProvider
...
 
+            OpenTelemetrySdkEventSource.Log.TracerProviderSdkEvent("Building TracerProvider.");
+
             var state = new TracerProviderBuilderState(serviceProvider);
 
             TracerProviderBuilderServiceCollectionHelper.InvokeRegisteredConfigureStateCallbacks(
                 serviceProvider,
                 state);
 
+            StringBuilder processorsAdded = new StringBuilder();
```
Comment (cijothomas) [inline src/OpenTelemetry/Trace/TracerProviderSdk.cs:61]: please also include sampler, as it is often the most critical information.

**Review by cijothomas — APPROVED** [review]
LGTM except https://github.com/open-telemetry/opentelemetry-dotnet/pull/3746/files#r996169712

Its okay to have it as follow up. Let us know if you can add it to this one, or separate one.

**Comment by cijothomas** [comment]
> 

lets merge. this is already an improvement! Could you follow up with a PR adding Samplers?

File: src/OpenTelemetry/Trace/TracerProviderSdk.cs (hunk 1 above)
Comment (ggoel) [reply src/OpenTelemetry/Trace/TracerProviderSdk.cs:61]: @cijothomas sure, will raise another PR for this

**Pull request merged**